.github/workflows/triage.yml

```diff
@@ -20,42 +19,105 @@ jobs:
-            --field project_number="$PROJECT_NUMBER" \
-            --raw-field repository_name="$REPOSITORY" \
+            --raw-field organization=$ORGANIZATION \
+            --field project_number=$PROJECT_NUMBER \
             --raw-field query='
-              query($owner: String!, $repository_name: String!, $project_number: Int!) {
-                repository(owner: $owner, name: $repository_name) {
-                  project(number: $project_number) {
-                    columns(first: 10) {
+              query($organization: String!, $project_number: Int!) {
+                organization(login: $organization){
+                  projectV2(number: $project_number) {
+                    id
+                    fields(first:20) {
                       nodes {
-                        id,
-                        name
+                        ... on ProjectV2Field {
+                          id
+                          name
+                        }
+                        ... on ProjectV2SingleSelectField {
+                          id
+                          name
+                          options {
+                            id
+                            name
+                          }
+                        }
                       }
                     }
                   }
                 }
               }' > project_columns.json
-          echo 'COLUMN_ID='$(jq '.data.repository.project.columns.nodes[] | select(.name == "Triage") | .id' project_columns.json) >> $GITHUB_ENV
-          echo 'COLUMN_NAME='$(jq '.data.repository.project.columns.nodes[] | select(.name == "Triage") | .name' project_columns.json) >> $GITHUB_ENV
+          echo 'PROJECT_ID='$(
+            jq \
+              '.data.organization.projectV2.id' \
+              project_columns.json
+          ) >> $GITHUB_ENV
+          echo 'STATUS_FIELD_ID='$(
+            jq \
+              '.data.organization.projectV2.fields.nodes[]
+              | select(.name== "Status")
+              | .id' \
+              project_columns.json
+          ) >> $GITHUB_ENV
+          echo 'TRIAGE_COLUMN_ID='$(
+            jq --raw-output \
+              '.data.organization.projectV2.fields.nodes[]
+              | select(.name== "Status")
+              | .options[]
+              | select(.name=="To Triage")
+              | .id' \
+              project_columns.json
+          ) >> $GITHUB_ENV
 
-      - name: Add issue#${{ github.event.issue.number }} to Triage column
+      - name: Add issue#${{ github.event.issue.number }} to project
         env:
           ISSUE_ID: ${{ github.event.issue.node_id }}
         run: |
-          echo "Adding issue: ${{ github.event.issue.number }} to column $COLUMN_NAME in project $PROJECT_NUMBER"
-          if [[ $DRY_RUN == "false" ]]; then
+          echo "Adding issue: ${{ github.event.issue.number }} to project $PROJECT_NUMBER"
+          if [[ $DRY_RUN == 'false' ]]; then
```

Comment:
Shouldn't this be comparing equality to `true`?

Reply:
No, if it is `true` then it will execute the `mutations` which is not what we want the the `DRY_RUN` mode. In such mode it will only query the project data and echo some debug information. These information are not that helpful I would argue, but I don't know if there is a way to actually dry-run a mutation in the GH graphql API.

Also, thinking now here, maybe `DEBUG` is better than `DRY_RUN`, since the action will not be actually running the commands in such mode.

Reply:
Right. I was mistakenly assuming that this PR also wanted to run the automation with `DRY_RUN` off.

Reply:
Ah, I see. I think that for this one it is better to go for debug first because we need to check the token permissions as well.
I already tested it in a personal repo and the mechanic of the action is working, but there is always some minor config details that may be different in each organization/project/repo, so it is better to take some extra care on this one.